app/models/sample.rb

```diff
@@ -437,11 +437,18 @@ def deletable?(user)
 
   def self.public_samples
     joins(:project)
-      .where("(projects.public_access = 1 or
+      .where("(projects.public_access = 1 OR
               DATE_ADD(samples.created_at, INTERVAL projects.days_to_keep_sample_private DAY) < ?)",
              Time.current)
   end
 
+  def self.private_samples
+    joins(:project)
+      .where("(projects.public_access = 0 AND
+              DATE_ADD(samples.created_at, INTERVAL projects.days_to_keep_sample_private DAY) >= ?)",
```

Comment:
This implies that once a sample is old enough it *leaves* the library and becomes public. Users may be surprised when they can't find something in their library that was there yesterday. It also seems strange that you may not be able to see all samples in a project because they could be divided between public and private.

Reply:
* Users are notified in app when samples go public
* There is a project/task to send emails on visibility changes
* It is part of our data policy

* The public samples of private projects are used to display on 
heatmaps, maps, phylo trees etc, so it is not weird that samples and projects can have different visibilities. Having said that, our access control mechanism  has some caveats both logic and UX, so eventually we will have to revisit it.